cli/cmd/plugin/diagnostics/e2e-test/e2e_test.go

```diff
@@ -0,0 +1,187 @@
+		signal = failConst
+	}
+
+	err = os.Chdir(tceRepoPath + "/cli/cmd/plugin/unmanaged-cluster")
+	if err != nil {
+		t.Errorf("error while changing directory to unmanged-cluster: %v", err)
+		signal = failConst
+	}
+	// creating Unmanaged cluster
+	err = cliRunner("go", nil, "run", ".", "create", clusterName)
+	if err != nil {
+		t.Errorf("Error while Unmanaged Cluster creation: %v", err)
+		signal = failConst
+	}
+
+	if signal == failConst {
+		fmt.Println("-------------------------------", string(colorRed), "Setup for diagnostic e2e-test is failed see abov logs", string(colorReset), "---------------------------------------------")
+	} else {
+		fmt.Println("-------------------------------", string(colorGreen), "Setup for diagnostic e2e-test is completed", string(colorReset), "---------------------------------------------")
+	}
+}
+
+func TestDiagnostic(t *testing.T) {
+	signal := passConst
+	tempDir, err := os.MkdirTemp("", "diagnosticE2Etest")
+	if err != nil {
+		t.Errorf("error while creating temp directory: %v", err)
+		signal = failConst
+	}
+
+	err = os.Chdir(tanzuDiagnosticPluginDir)
+	if err != nil {
+		t.Errorf("error while changing directory to Diagnostic plugin: %v", err)
+		signal = failConst
+	}
+
+	diagnostic := []string{"run", ".", "collect",
+		"--bootstrap-cluster-name", clusterName,
+		"--management-cluster-kubeconfig", home + "/.kube/config",
+		"--management-cluster-context", clusterKubeContext,
+		"--management-cluster-name", clusterName,
+		"--workload-cluster-infra", "docker",
+		"--workload-cluster-kubeconfig", home + "/.kube/config",
+		"--workload-cluster-context", clusterKubeContext,
+		"--workload-cluster-name", clusterName,
+		"--unmanaged-cluster-kubeconfig", home + "/.kube/config",
+		"--unmanaged-cluster-context", clusterKubeContext,
+		"--unmanaged-cluster-name", clusterName,
+		"--output-dir", tempDir}
+
+	// Collecting logs for all the cluster
+	err = cliRunner("go", nil, diagnostic[:]...)
+	if err != nil {
+		t.Errorf("error while collectiong diagnostic %v", err)
+		signal = failConst
+	}
+
+	// Checking all the log files
+	diagnosticFiles := [4]string{"/bootstrap." + clusterName + ".diagnostics.tar.gz",
+		"/management-cluster." + clusterName + ".diagnostics.tar.gz",
+		"/workload-cluster." + clusterName + ".diagnostics.tar.gz",
+		"/unmanaged-cluster." + clusterName + ".diagnostics.tar.gz"}
+	for _, element := range diagnosticFiles {
+		_, err = os.Stat(tempDir + element)
+		if err != nil {
+			t.Errorf("error while looking for %v %v", element, err)
+			signal = failConst
+		}
+	}
+
+	// cleaning th temp directory
+	err = os.RemoveAll(tempDir)
+	if err != nil {
+		t.Error(err)
+		signal = failConst
+	}
+
+	// deleting Unmanaged cluster
+	err = os.Chdir(tceRepoPath + "/cli/cmd/plugin/unmanaged-cluster")
+	if err != nil {
+		t.Errorf("error while changing directory to unmanged-cluster: %v", err)
+		signal = failConst
+	}
+
+	err = cliRunner("go", nil, "run", ".", "delete", clusterName)
```

Comment:
Same here. We could easily import the `unmanaged-cluster` tanzu package and call `Manager.Delete` with the cluster name.